### ISSUE TITLE
Error when deleting a rule

### ISSUE THREAD

**Issue by sdonovan1985** [atlanticwave-sdx/atlanticwave-proto#39]
This occurs when deleting 1 or all rules. Not sure exactly what's going on:

[2017-10-09 17:54:11,350] ERROR in app: Exception on /rule/all/ [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1988, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1641, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1544, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1639, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1625, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/atlanticwave-proto/sdxctlr/RestAPI.py", line 483, in get_rules
    RuleManager.instance().remove_all_rules(flask_login.current_user.id)
  File "/atlanticwave-proto/sdxctlr/RuleManager.py", line 183, in remove_all_rules
    self.remove_rule(rule['hash'], user)
  File "/atlanticwave-proto/sdxctlr/RuleManager.py", line 177, in remove_rule
    self._rm_rule_from_db(rule)
  File "/atlanticwave-proto/sdxctlr/RuleManager.py", line 376, in _rm_rule_from_db
    self._remove_rule(rule.get_rule_hash())
  File "/atlanticwave-proto/sdxctlr/RuleManager.py", line 416, in _remove_rule
    rule = pickle.loads(str(table_entry['rule']))
TypeError: 'NoneType' object has no attribute '__getitem__'

**Comment by sdonovan1985** [comment]
https://github.com/atlanticwave-sdx/atlanticwave-proto/blob/master/configuration/helloworld/README.md will need to be updated when this is fixed.

**Comment by sdonovan1985** [comment]
Fixed this issue, but ran into a new one on the Local Controller. I was able to delete individual rules correctly, but I tried "delete all" and something went awry in the LC.

There were two rules running on the helloworld topology, one connecting the DTNs and one connecting ports 1 and 3.

This only happens when I ran a "delete all". Unless I'm mistaken, delete-all is just a for loop, deleting them individually.

Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/atlanticwave-proto/localctlr/RyuTranslateInterface.py", line 273, in main_loop
    self.remove_rule(datapath, event)
  File "/atlanticwave-proto/localctlr/RyuTranslateInterface.py", line 1204, in remove_rule
    for rule in swrules:
TypeError: 'NoneType' object is not iterable

Sent     : RM_RULE:(1, 3)
Receiving Command on sdx_connection
Received : RM_RULE:(1, 6)
Sent     : RM_RULE:(1, 6)
Receiving Command on sdx_connection
Received : RM_RULE:(1, 7)
Sent     : RM_RULE:(1, 7)



Oh, this is fun.

So what's happening is that it's trying to delete some rules that were automatically installed that shouldn't be deleted. This can be produced by starting up the SDXController in switchless mode (`python SDXController.py -m ../configuration/helloworld/helloworld.manifest -N`), and clicking "Delete All Rules" on the view all rules page.

Working on fixing this.
